go/arrow/scalar/scalar_test.go

```diff
@@ -1416,3 +1427,56 @@ func TestUnionScalars(t *testing.T) {
 	suite.Run(t, new(SparseUnionSuite))
 	suite.Run(t, new(DenseUnionSuite))
 }
+
+func TestRunEndEncodedGetScalar(t *testing.T) {
+	mem := memory.NewCheckedAllocator(memory.DefaultAllocator)
+	defer mem.AssertSize(t, 0)
+
+	runEnds, _, _ := array.FromJSON(mem, arrow.PrimitiveTypes.Int32, strings.NewReader(`[100, 200, 300, 400, 500]`))
+	defer runEnds.Release()
+
+	values, _, _ := array.FromJSON(mem, arrow.BinaryTypes.String, strings.NewReader(`["Hello", "beautiful", "world", "of", "RLE"]`))
+	defer values.Release()
+
+	reeArray := array.NewRunEndEncodedArray(runEnds, values, 500, 0)
+	defer reeArray.Release()
```

Comment:
Is there a plan to implement FromJSON for REE? (Possibly using an analogous encoding to unions like `[run_length, value]`?)

Reply:
`FromJSON` was already implemented for REE, and like dictionary arrays, REE arrays are expected to be expanded in the JSON. Processing the JSON will encode it into run-end-encoding

Reply:
I guess what I mean is, why are we still constructing the array manually here? But it doesn't really matter